manifests/client.pp

```diff
@@ -32,27 +32,48 @@
         $service_enable = false
       }
     }
+    case $::osfamily {
+      'windows': {
+        $service_name     = 'sensu-client'
+        $service_path     = undef
+        $service_provider = undef
+        file { 'C:/opt/sensu/bin/sensu-client.xml':
+          ensure  => file,
+          content => template("${module_name}/sensu-client.erb"),
+        }
 
-    if $::osfamily == 'windows' {
-
-      file { 'C:/opt/sensu/bin/sensu-client.xml':
-        ensure  => file,
-        content => template("${module_name}/sensu-client.erb"),
+        exec { 'install-sensu-client':
+          provider => 'powershell',
+          command  => "New-Service -Name sensu-client -BinaryPathName c:\\opt\\sensu\\bin\\sensu-client.exe -DisplayName 'Sensu Client' -StartupType Automatic",
+          unless   => 'if (Get-Service sensu-client -ErrorAction SilentlyContinue) { exit 0 } else { exit 1 }',
+          before   => Service[$::sensu::service_name],
+          require  => File['C:/opt/sensu/bin/sensu-client.xml'],
+        }
       }
+      'Darwin': {
+        $service_path     = '/Library/LaunchDaemons/org.sensuapp.sensu-client.plist'
+        $service_provider = 'launchd'
 
-      exec { 'install-sensu-client':
-        provider => 'powershell',
-        command  => "New-Service -Name sensu-client -BinaryPathName c:\\opt\\sensu\\bin\\sensu-client.exe -DisplayName 'Sensu Client' -StartupType Automatic",
-        unless   => 'if (Get-Service sensu-client -ErrorAction SilentlyContinue) { exit 0 } else { exit 1 }',
-        before   => Service['sensu-client'],
-        require  => File['C:/opt/sensu/bin/sensu-client.xml'],
+        file {$service_path:
+          ensure => file,
+          owner  => 'root',
+          group  => 'wheel',
+          mode   => '0755',
+          before => Service[$::sensu::service_name],
+        }
+      }
+      default: {
+        $service_path     = undef
+        $service_provider = undef
       }
     }
 
-    service { 'sensu-client':
```

Comment:
The title should remain `sensu-client` though we can use the `name` attribute to set the actual name.